### PR TITLE
[Compute] BREAKING CHANGE: az vm/vmss create: New VM from existing disk or image

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

Close: #23476

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 1]
```diff
@@ -602,6 +602,23 @@ def trusted_launch_warning_log(namespace, generation_version, features):
...
+
+    from ._constants import UPGRADE_SECURITY_HINT
+    security_type = disk_security_profile.security_type if hasattr(disk_security_profile, 'security_type') else None
+    if security_type == 'TrustedLaunch':
+        if namespace.enable_secure_boot is None:
+            namespace.enable_secure_boot = True
+        if namespace.enable_vtpm is None:
+            namespace.enable_vtpm = True
+        if namespace.security_type is None:
+            namespace.security_type = 'TrustedLaunch'
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:617]: It is mentioned in the feature request #23476 Scenario 1:
OS Disk securityProfile has 1-1 mapping with VM, i.e., OS Disk enabled with Trusted Launch **can only** deploy VM with Trusted Launch.

I think namespace.security_type shoule be set to TrustedLaunch directly without if statement. Hi @AjKundnani could you please help confirm this?
```suggestion
        namespace.security_type = 'TrustedLaunch'
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -1404,7 +1404,38 @@ def _validate_generation_version_and_trusted_launch(cmd, namespace):
         generation_version = attach_os_disk_info.hyper_v_generation if hasattr(attach_os_disk_info,
                                                                                'hyper_v_generation') else None
         features = attach_os_disk_info.features if hasattr(attach_os_disk_info, 'features') else None
+        disk_security_profile = attach_os_disk_info.security_profile if hasattr(attach_os_disk_info,
+                                                                               'security_profile') else None
         trusted_launch_warning_log(namespace, generation_version, features)
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1409]: This warning message may should be removed
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 3]
```diff
@@ -602,6 +602,23 @@ def trusted_launch_warning_log(namespace, generation_version, features):
             logger.warning(UPGRADE_SECURITY_HINT)
 
 
+def validate_vm_disk_trusted_launch(namespace, disk_security_profile):
+    if disk_security_profile is None:
+        return
+
+    from ._constants import UPGRADE_SECURITY_HINT
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:609]: The warning message mentioned in the feature request is:
`Consider upgrading security for your workloads using Azure Trusted Launch VMs. To know more about Trusted Launch, please visit https://aka.ms/TrustedLaunch`

Hence the link in the UPGRADE_SECURITY_HINT field should be changed from "https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch" to "https://aka.ms/TrustedLaunch".

**Comment by yanzhudd** [comment]
For creating VM from existing shared image (Azure Compute Gallery) or managed image using az vm create --image (Scenario 2), corresponding logic should be changed ([_validators.py L1351](https://github.com/ZengTaoxu/azure-cli/blob/eef61524a554957a1fc197ffa527585180dd5735/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L1351) and [_validators.py L1359](https://github.com/ZengTaoxu/azure-cli/blob/eef61524a554957a1fc197ffa527585180dd5735/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L1359)).

You can refer to a similar logic with the feature #23614 Scenario 2.
https://github.com/Azure/azure-cli/issues/23614#issuecomment-1765645238
PR: https://github.com/Azure/azure-cli/pull/27620/files

**Comment by yanzhudd** [comment]
Scenario 3 is similar with Scenario 2.

**Pull request closed without merging**